### ISSUE TITLE
Missing python inference exmple in documentation

### ISSUE THREAD

**Issue by windelbouwman** [PyCQA/astroid#480]
This package looks great, but I find the documentation somewhat hard to read. I would like to add a gentle introduction page with an example of some parsed python code and then the best way to infer some of the types used in this code.

I am willing to do this myself, but I am no expert in this library. How to proceed?

**Comment by PCManticore** [comment]
That's true, we're definitely lacking in the documentation department, although things improved significantly since we added #458 . You could find some tips for each kind of node and about general inference in the code base, as the documentation for the corresponding nodes. Here's an example for instance: https://github.com/PyCQA/astroid/blob/master/astroid/node_classes.py#L1142. Another place you could take a look as well is in pylint's documentation, especially in the Writing a checker section: https://pylint.readthedocs.io/en/latest/how_tos/custom_checkers.html. Other than these two, usually reading the code and seeing how something is done is the best way to figure out how a pattern works. Hope this helps, but if you need any help, you can find some extra help on IRC at #pylint as well.
